layers/water-material.js

```diff
@@ -0,0 +1,346 @@
+
+        uniform float uTime;
+        uniform sampler2D tDepth;
+        uniform sampler2D foamTexture;
+        uniform float cameraNear;
+        uniform float cameraFar;
+        uniform vec2 resolution;
+        uniform sampler2D tDistortion;
+
+        uniform sampler2D waterNormalTexture;
+        uniform samplerCube cubeMap;
+      
+        varying vec3 vWorldPosition;
+        varying vec3 vNormal;
+        varying vec4 vUv;
+        varying vec3 vPos;
+
+        const float TAU = 2. * 3.14159265;
+
+        const vec3 eyePosition = vec3(0.7579705245610807, 0.6382203660633491, 0.1347421546456965);
+
+        // cosine gradient 
+        const vec4 phases = vec4(0.28, 0.50, 0.07, 0);
+        const vec4 amplitudes = vec4(4.02, 0.34, 0.65, 0);
+        const vec4 frequencies = vec4(0.00, 0.48, 0.08, 0);
+        const vec4 offsets = vec4(0.00, 0.17, 0.00, 0);
+
+        vec4 cosine_gradient(float x, vec4 phase, vec4 amp, vec4 freq, vec4 offset){
+          phase *= TAU;
+          x *= TAU;
+
+          return vec4(
+            offset.r + amp.r * 0.5 * cos(x * freq.r + phase.r) + 0.5,
+            offset.g + amp.g * 0.5 * cos(x * freq.g + phase.g) + 0.5,
+            offset.b + amp.b * 0.5 * cos(x * freq.b + phase.b) + 0.5,
+            offset.a + amp.a * 0.5 * cos(x * freq.a + phase.a) + 0.5
+          );
+        }
+
+        float getDepth(const in vec2 screenPosition) {
+          return unpackRGBAToDepth(texture2D(tDepth, screenPosition));
+        }
+
+        float getViewZ(const in float depth) {
+          return perspectiveDepthToViewZ(depth, cameraNear, cameraFar);
+        }  
+
+        float getDepthFade(float fragmentLinearEyeDepth, float linearEyeDepth, float depthScale, float depthFalloff) {
+          return pow(saturate(1. - (fragmentLinearEyeDepth - linearEyeDepth) / depthScale), depthFalloff);
+        }
+
+        vec4 cutout(float depth, float alpha) {
+          return vec4(ceil(depth - saturate(alpha)));
+        }
+
+        vec4 getNoise( vec2 uv ) {
+          vec2 uv0 = ( uv / 103.0 ) - vec2(uTime / 17.0, uTime / 29.0);
+          vec2 uv1 = uv / 107.0 + vec2( uTime / -19.0, uTime / 31.0 );
+          vec2 uv2 = uv / vec2( 8907.0, 9803.0 ) - vec2( uTime / 101.0, uTime / 97.0 );
+          vec2 uv3 = uv / vec2( 1091.0, 1027.0 ) + vec2( uTime / 109.0, uTime / -113.0 );
+          vec4 noise = texture2D( waterNormalTexture, uv0 ) +
+            texture2D( waterNormalTexture, uv1 ) +
+            texture2D( waterNormalTexture, uv2 ) +
+            texture2D( waterNormalTexture, uv3 );
+          return noise * 0.5 - 1.0;
+        }
+        
+        void main() {
+          vec2 screenUV = gl_FragCoord.xy / resolution;
+
+          float fragmentLinearEyeDepth = getViewZ(gl_FragCoord.z);
+          float linearEyeDepth = getViewZ(getDepth(screenUV));
+
+          if (vPos.y <= 0.01) { // ocean shader, for temporary. TODO: We should have attribute to indicate the water type(ocean, river, waterfall, etc.,)
```

Comment:
Okay, this is easily doable.
Before we merge this PR I'll add the liquid type to the attributes.
Further, we should rename all of the `water` keywords to be `liquid`, since that's what it really is.

Reply:
> Before we merge this PR I'll add the liquid type to the attributes.
Thanks, I also need flow direction if you can

Reply:
[change water to liquid](https://github.com/webaverse/heightfield/pull/19/commits/6627d439c51ee1a22dc2e4c6c059b33f60b2adc4)